e2e/internal/e2e/singularitycmd.go

```diff
@@ -290,12 +290,15 @@ func WithDir(dir string) SingularityCmdOp {
 	}
 }
 
-// WithPrivileges sets whether a singularity command must be
-// executed with privileges or not. PreRun, InRun, PostRun
-// are also executed with privileges.
-func WithPrivileges(privileged bool) SingularityCmdOp {
+// WithProfile sets the Singularity execution profile, this
```

Comment:
My preference goes for always provide a profile (to avoid the question "huh wait what is the default behavior ?" when reading code. As profile in this PR are redundant with `WithPrivileges`, we now have something like when using profile :

`UserProfile` -> `WithPrivileges(false)`
`RootProfile` -> `WithPrivileges(true)`
`UserNamespaceProfile` -> `WithPrivileges(false)` inject `--userns` option for commands `shell,test,exec,run,instance start`
`FakerootProfile` -> `WithPrivileges(false)` inject `--fakeroot` option for commands `shell,test,exec,run,instance start,build`